server/sonar-server/src/main/java/org/sonar/server/component/ComponentUpdater.java

```diff
@@ -100,6 +103,17 @@ private ComponentDto createRootComponent(DbSession session, NewComponent newComp
       .setPrivate(newComponent.isPrivate())
       .setCreatedAt(new Date(system2.now()));
     dbClient.componentDao().insert(session, component);
+
+    BranchDto branch = new BranchDto()
```

Comment:
Warning, this class is used by Governance to create View.
And we should not create main branch on views.

I think the creation of the main branch should only be done during the analysis, not during the provisioning.